### PR TITLE
make sure train_model_and_embed_images returns everything

### LINKED ISSUE #696: _lightly_cli is missing it's return statement
as a side effect `train_model_and_embed_images` is not working as expected.

### PR DESCRIPTION
**Pull request by MalteEbner**
## Description
- wrote unittest reproducing the issue
- fixed it in the `train_model_and_embed_images()` function than now uses the `_train_cli()` and `_embed_cli()` instead of the `_lightly_cli()`
- added comments to the `_embed_cli()` to make the return type much clearer
- added a unittest to the lightly-cli to prevent lines from becoming uncovered

closes #696

### CODE REVIEW

File: lightly/cli/embed_cli.py [hunk 1]
```diff
@@ -24,7 +26,17 @@
 from lightly.cli._helpers import cpu_count
 
 
-def _embed_cli(cfg, is_cli_call=True):
+def _embed_cli(cfg, is_cli_call=True) -> \
+    Union[
+        Tuple[List[np.ndarray], List[int], List[str]],
```
Comment (philippmwirth) [inline lightly/cli/embed_cli.py:31]: It's not a list of array, it's just an np.ndarray

File: lightly/core.py [hunk 2]
```diff
@@ -73,7 +76,9 @@ def _add_kwargs(cfg, kwargs):
     return cfg
 
 
-def train_model_and_embed_images(config_path: str = None, **kwargs):
+def train_model_and_embed_images(config_path: str = None, **kwargs) -> Tuple[
+    List[np.ndarray], List[int], List[str]
```
Comment (philippmwirth) [inline lightly/core.py:80]: same here, or am I wrong?

File: tests/cli/test_cli_magic.py [hunk 3]
```diff
@@ -48,13 +48,29 @@ def parse_cli_string(self, cli_words: str):
...
+                pass
+
+            key_subparts = key.split('.')
+            if len(key_subparts) == 1:
+                self.cfg[key] = value
+            elif len(key_subparts) == 2:
+                self.cfg[key_subparts[0]][key_subparts[1]] = value
+            else:
+                raise ValueError(
+                    f"Keys with more than 2 subparts are not supported,"
```
Comment (philippmwirth) [inline tests/cli/test_cli_magic.py:63]: Above you used single quotes and here you use double quotes. I would always use single quotes.

**Review by philippmwirth — COMMENTED** [review]
nice one! just a few remarks

File: lightly/cli/embed_cli.py (hunk 1 above)
Comment (MalteEbner) [reply lightly/cli/embed_cli.py:31]: Thanks, good that we add the typehints :)

File: tests/cli/test_cli_magic.py [hunk 4]
```diff
@@ -60,8 +60,8 @@ def parse_cli_string(self, cli_words: str):
                 self.cfg[key_subparts[0]][key_subparts[1]] = value
             else:
                 raise ValueError(
-                    f"Keys with more than 2 subparts are not supported,"
-                     "but you entered {key}."
+                    f'Keys with more than 2 subparts are not supported,'
+                     'but you entered {key}.'
```
Comment (philippmwirth) [inline tests/cli/test_cli_magic.py:64]: I just noticed that the `f` is in the wrong place here

**Review by philippmwirth — CHANGES_REQUESTED** [review]

**Review by philippmwirth — APPROVED** [review]

**Pull request merged**